sdk/core/azure-core/src/main/java/com/azure/core/util/logging/LoggingEventBuilder.java

```diff
@@ -205,14 +255,30 @@ public Throwable log(Throwable throwable) {
         return throwable;
     }
 
+    /**
+     * Logs the {@link RuntimeException} and returns it to be thrown.
+     * This API covers the cases where a checked exception type needs to be thrown and logged.
+     *
+     * @param runtimeException RuntimeException to be logged and returned.
+     * @return The passed {@link RuntimeException}.
+     * @throws NullPointerException If {@code runtimeException} is {@code null}.
+     */
+    public RuntimeException log(RuntimeException runtimeException) {
```

Comment:
Do we need this overload since we already have `log(Throwable throwable)` method?

Reply:
yes, we need it for similar reasons we have one on `ClientLogger`: when we do `throw log(RuntimeException)` we don't have to add checked exception or cast the exception type.